OBUSPA: Disable MQTT support
  
MQTT support has been removed because OBUSPA requires a newer version
of libmosquitto than is currently available, and there is an existing
buld which requires the older libmosquitto. The newer version of
libmosquitto supports MQTTv5, whilst the older version supports only
MQTTv3.1. OBUSPA requires MQTTv5 support.

This patch could be removed if either the existing build moved to the
newer libmosquitto, or BitBake could be persuaded to build both
versions of libmosquitto and link the requisite version in each case.
  
Date: Fri, 19 Feb 2021 00:00:00 +0000
From: Richard Holme <richard.holme@commscope.com>
Subject: [PATCH] Disable MQTT support
Source: CommScope
License: BSD-3-Clause
Upstream-Status: Inappropriate [disable feature]
Signed-off-by: Richard Holme <richard.holme@commscope.com>




diff -rupN old/configure.ac new/configure.ac
--- old/configure.ac	2020-09-29 16:40:39.000000000 +0100
+++ new/configure.ac	2020-10-26 10:42:16.192223330 +0000
@@ -22,7 +22,6 @@ PKG_CHECK_MODULES([openssl], [openssl])
 PKG_CHECK_MODULES([sqlite3], [sqlite3])
 PKG_CHECK_MODULES([libcurl], [libcurl])
 PKG_CHECK_MODULES([zlib], [zlib])
-PKG_CHECK_MODULES([libmosquitto], [libmosquitto])
 
 # Check which flavour of strerror_r is available on the target
 AC_FUNC_STRERROR_R
diff -rupN old/Makefile.am new/Makefile.am
--- old/Makefile.am	2020-09-29 16:40:39.000000000 +0100
+++ new/Makefile.am	2020-10-26 10:44:29.874731255 +0000
@@ -83,14 +83,11 @@ SOURCES = src/core/main.c \
                     src/core/coap_common.c \
                     src/core/coap_client.c \
                     src/core/coap_server.c \
-                    src/core/uri.c \
-                    src/core/mqtt.c \
-                    src/core/device_mqtt.c
+                    src/core/uri.c
 
 obuspa_SOURCES = $(SOURCES)
-obuspa_CPPFLAGS = $(openssl_CFLAGS) $(sqlite3_CFLAGS) $(libcurl_CFLAGS) $(zlib_CFLAGS) $(libmosquitto_CFLAGS)
+obuspa_CPPFLAGS = $(openssl_CFLAGS) $(sqlite3_CFLAGS) $(libcurl_CFLAGS) $(zlib_CFLAGS)
 obuspa_CPPFLAGS += -DENABLE_COAP
-obuspa_CPPFLAGS += -DENABLE_MQTT
 obuspa_CPPFLAGS += -DOBUSPA_LOCAL_STATE_DIR=\"$(OBUSPA_LOCAL_STATE_DIR)\"
 obuspa_CPPFLAGS +=  $(AM_CPPFLAGS) \
                       -Werror \
@@ -105,7 +102,7 @@ obuspa_CPPFLAGS +=  $(AM_CPPFLAGS) \
 obuspa_LDFLAGS = -rdynamic
 
 obuspa_LDADD = -lm -ldl -lpthread -lrt
-obuspa_LDADD += $(openssl_LIBS) $(sqlite3_LIBS) $(libcurl_LIBS) $(zlib_LIBS) $(libmosquitto_LIBS)
+obuspa_LDADD += $(openssl_LIBS) $(sqlite3_LIBS) $(libcurl_LIBS) $(zlib_LIBS)
 
 obuspa_LDFLAGS += -Wl,-rpath=/usr/local/lib
 
